--- /usr/share/awesome/lib/eminent.lua	2013-01-03 05:31:15.000000000 +0900
+++ eminent.lua	2013-01-03 09:38:23.495794380 +0900
@@ -41,7 +41,7 @@
     viewidx = awful.tag.viewidx,
 
     taglist = awful.widget.taglist.new,
-    label = awful.widget.taglist.label.all,
+    filter = awful.widget.taglist.filter.all,
 }
 
 -- Return tags with stuff on them, mark others hidden
@@ -100,26 +100,29 @@
 end
 
 -- Taglist label functions
-awful.widget.taglist.label.all = function (t, args)
+awful.widget.taglist.filter.all = function (t, args)
     if t.selected or #t:clients() > 0 then
-        return orig.label(t, args)
+        return orig.filter(t, args)
     end
 end
 
 
 -- Update hidden status
-local function uc(c) gettags(c.screen) end
-local function ut(s, t) gettags(s.index) end
-
-capi.client.add_signal("unmanage", uc)
-capi.client.add_signal("new", function(c)
-    c:add_signal("property::screen", uc)
-    c:add_signal("tagged", uc)
-    c:add_signal("untagged", uc)
-end)
-
-for screen=1, capi.screen.count() do
-    awful.tag.attached_add_signal(screen, "property::selected", uc)
-    capi.screen[screen]:add_signal("tag::attach", ut)
-    capi.screen[screen]:add_signal("tag::detach", ut)
-end
+-- local function uc(c)
+--     naughty.notify({title="uc",text=c.screen})
+--     gettags(c.screen) 
+-- end
+-- local function ut(s, t) gettags(s.index) end
+-- 
+-- capi.client.connect_signal("unmanage", uc)
+-- capi.client.connect_signal("new", function(c)
+--     c:connect_signal("property::screen", uc)
+--     c:connect_signal("tagged", uc)
+--     c:connect_signal("untagged", uc)
+-- end)
+-- 
+-- for screen=1, capi.screen.count() do
+--     awful.tag.attached_connect_signal(screen, "property::selected", uc)
+--     capi.screen[screen]:connect_signal("tag::attach", ut)
+--     capi.screen[screen]:connect_signal("tag::detach", ut)
+-- end
